src/Microsoft.Health.Fhir.SqlServer/Features/Schema/SchemaInitializer.cs

```diff
@@ -0,0 +1,143 @@
+using Microsoft.Health.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.SqlServer.Configs;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Schema
+{
+    /// <summary>
+    /// EXPERIMENTAL - Initializes the sql schema and brings the schema up to the min supported version.
+    /// The purpose of this it to enable easy scenarios during development and will likely be removed later.
+    /// </summary>
+    public class SchemaInitializer : IStartable
+    {
+        private readonly SqlServerDataStoreConfiguration _sqlServerDataStoreConfiguration;
+        private readonly SchemaUpgradeRunner _schemaUpgradeRunner;
+        private readonly SchemaInformation _schemaInformation;
+        private readonly ILogger<SchemaInitializer> _logger;
+
+        public SchemaInitializer(SqlServerDataStoreConfiguration sqlServerDataStoreConfiguration, SchemaUpgradeRunner schemaUpgradeRunner, SchemaInformation schemaInformation, ILogger<SchemaInitializer> logger)
+        {
+            EnsureArg.IsNotNull(sqlServerDataStoreConfiguration, nameof(sqlServerDataStoreConfiguration));
+            EnsureArg.IsNotNull(schemaUpgradeRunner, nameof(schemaUpgradeRunner));
+            EnsureArg.IsNotNull(schemaInformation, nameof(schemaInformation));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _sqlServerDataStoreConfiguration = sqlServerDataStoreConfiguration;
+            _schemaUpgradeRunner = schemaUpgradeRunner;
+            _schemaInformation = schemaInformation;
+            _logger = logger;
+        }
+
+        private void Initialize()
+        {
+            if (!CanInitialize())
+            {
+                return;
+            }
+
+            if (!IsSchemaTableInitialized())
+            {
+                CreateSchemaTable();
+            }
+
+            SetCurrentSchemaVersion();
+
+            if (_schemaInformation.Current == null || _schemaInformation.Current < _schemaInformation.Min)
+            {
+                int current = (int?)_schemaInformation.Current ?? 0;
+
+                for (int i = current + 1; i <= (int)_schemaInformation.Min; i++)
+                {
+                    _schemaUpgradeRunner.ApplySchema(i);
+                }
+            }
+        }
+
+        private void CreateSchemaTable()
+        {
+            using (var connection = new SqlConnection(_sqlServerDataStoreConfiguration.ConnectionString))
+            {
+                var command = new SqlCommand("CREATE TABLE SchemaTable ( Version int PRIMARY KEY, Status varchar(10) )", connection);
+
+                connection.Open();
+                command.ExecuteNonQuery();
+            }
+        }
+
+        private bool IsSchemaTableInitialized()
+        {
+            using (var connection = new SqlConnection(_sqlServerDataStoreConfiguration.ConnectionString))
+            {
+                var command = new SqlCommand("SELECT count(*) FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = 'SchemaTable'", connection);
```

Comment:
Schema is an overloaded term in SQL. I think we should call this Version or SchemaVersion.